### ISSUE TITLE
Implement Firestore TTL Policies for log/inquiry entries maintenance 

### ISSUE THREAD

**Issue by jessegduke** [dora-team/dora.dev#329]
We should use [time-to-live (TTL) policies](https://firebase.google.com/docs/firestore/ttl) to automatically remove stale data from Firestore database. 

The [Send Mail Extension already supports such protections](https://extensions.dev/extensions/firebase/firestore-send-email)  in their data structures.  We just need to implement such policies.

Ideally: Implement via Terraform if supported - and that can be the start of the Tf efforts for IaC management.

**Comment by jessegduke** [comment]
FYI @davidstanke.  I have added a document TTL Policy to `doradotdev-staging` firestore db (and I have started building Infra as Code resources to support future efforts).  

I will monitor over the next week to confirm success.

The `email-log` collection is "good to go", but we will need to adjust the `email-inquiry` collection schema to add a date field for the policy to act upon.  I will create a new issue and capture as an enhancement to the existing cloud functions and assign to myself.
